### PR TITLE
Add right click option for reset character and cam controls

### PR DESCRIPTION
**Pull request by amzn-rhhong**
https://user-images.githubusercontent.com/69218254/155086838-3dad2e29-1de8-45b0-bb75-559f1c4c2071.mp4

### CODE REVIEW

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.cpp [hunk 1]
```diff
@@ -251,13 +251,19 @@ namespace EMStudio
         skyBoxFeatureProcessorInterface->SetCubemapRotationMatrix(rotationMatrix);
 
         // Reset ground entity
-        AZ::Transform groundTransform = AZ::Transform::CreateIdentity();
-        AZ::TransformBus::Event(m_groundEntity->GetId(), &AZ::TransformBus::Events::SetLocalTM, groundTransform);
+        AZ::Transform identityTransform = AZ::Transform::CreateIdentity();
+        AZ::TransformBus::Event(m_groundEntity->GetId(), &AZ::TransformBus::Events::SetLocalTM, identityTransform);
```
Comment (amzn-jillich) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.cpp:255]: You might just call the UpdateGroundplane() function here after resetting the actor positions below.

**Review by amzn-jillich — APPROVED** [review]

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.cpp [hunk 2]
```diff
@@ -290,4 +290,75 @@ namespace EMStudio
...
+        }
+
+        QAction* resetAction = menu.addAction("Reset Character");
+        connect(resetAction, &QAction::triggered, this, [this]()
+            {
+                m_renderer->Reinit();
+                UpdateCameraViewMode(RenderOptions::CameraViewMode::DEFAULT);
+            });
+
+        menu.exec(event->globalPos());
```
Comment (hultonha) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.cpp:362]: suggestion: You might want to double check with the Editor team if we should use `exec` here or `popup`, I think `exec` is blocking and might not be the right thing to use here, but @cgalvan will know for sure

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.h [hunk 3]
```diff
@@ -8,13 +8,15 @@
...
 #include <QSettings>
+#include <QMouseEvent>
 #include <AtomToolsFramework/Viewport/RenderViewportWidget.h>
 #include <AzFramework/Viewport/CameraInput.h>
 
 #include <EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/RenderPlugin/ViewportPluginBus.h>
 #include <EMStudio/AnimViewportRequestBus.h>
 #include <Integration/Rendering/RenderFlag.h>
 
+
```
Comment (hultonha) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.h:19]: uber nit: Lose the extra line break

**Review by hultonha — APPROVED** [review]
LGTM with one suggestion to check about the context menu

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.cpp (hunk 2 above)
Comment (hultonha) [reply Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.cpp:362]: See `EditorContextMenu.cpp` as an example

Comment (amzn-rhhong) [reply Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.cpp:362]: yeah I will switch to use pop

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.h (hunk 3 above)
Comment (amzn-rhhong) [reply Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.h:19]: I thought we suppose to have double line break between include and actual code starts. The standard is a grey area probably?

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.cpp (hunk 1 above)
Comment (amzn-rhhong) [reply Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.cpp:268]: Change has not been merged yet. But yeah that's a good idea.

**Pull request merged**